### DISCUSSION TITLE
Side Panel Table Block shows all rows instead of specific

### DISCUSSION THREAD

**Discussion by japji316** [Budibase/budibase#9347] [Help] [answered]
Hi, 
When defining table row click action, the side panel shows all the rows in a table instead of showing just the row clicked on.
I have the Side Panel row id pointing to the table._id.
Is there a way to only show the clicked row?
The view link url also points to the same table._id and opens the respective row.
![Id1](https://user-images.githubusercontent.com/10275744/212564590-08b83a96-e037-4474-8f81-d71bdbe0cae1.PNG)
![Id2](https://user-images.githubusercontent.com/10275744/212564593-fc18e8f7-c4e6-4edd-91d9-fd59e81accec.PNG)

**Comment by melohagan** [comment]
Hey @japji316

I'm not sure what the issue is here. When you click the second row, it is showing the details for that row that was clicked, no?

**Comment by japji316** [comment]
Hey @melohagan , The issue is that it shows all the rows, but starts at Row ID 1, regardless of what row is clicked. The rest of the rows show but you have to scroll down to them.

**Reply by melohagan** [reply]
Oh that's odd, I've never seen that before. Would you be able to share your app export and I can take a look?

**Comment by japji316** [comment]
Sure, here is the App export.
https://we.tl/t-S6fZGCiHjs
<https://we.tl/t-S6fZGCiHjs?utm_campaign=TRN_TDL_05&utm_source=sendgrid&utm_medium=email&trk=TRN_TDL_05>
Specifically the "Jobs" page

**Comment by melohagan** [comment, marked as answer]
Thanks for the app export @japji316 I understand what the issue is now.

You have your side panels placed inside a Table component:
![Screenshot 2023-01-31 at 10 07 41](https://user-images.githubusercontent.com/101575380/215730514-2eb366fb-af1f-48dd-b7b2-eaa7fab4188d.png)

The table component is going to repeat any children for each row, thus your side panels are being repeated 4 times (once for each row).

To fix, simply move the Side Panels outside of your table component. Side panels are managed by state, so they can be placed at any level of your screen and still be opened.


**Fix**
![Screenshot 2023-01-31 at 10 09 36](https://user-images.githubusercontent.com/101575380/215730923-daaa1492-58ca-459a-a368-c67a33e1630f.png)

![Screenshot 2023-01-31 at 10 10 04](https://user-images.githubusercontent.com/101575380/215731019-afee876a-9142-4603-9af9-a6c49557b925.png)

**Comment by japji316** [comment]
Got it, thank you for the clarification. I recreated the table from a table
block component and got it pointing correctly.
